staging/vhost-device-console/src/backend.rs

```diff
@@ -0,0 +1,240 @@
+    #[clap(short = 'p', long, default_value = "path_to_console_socket")]
+    console_path: String,
+
+    /// Number of guests (sockets) to connect to.
+    #[clap(short = 'c', long, default_value_t = 1)]
+    socket_count: u32,
+}
+
+#[derive(PartialEq, Debug)]
+struct ConsoleConfiguration {
+    socket_path: String,
+    socket_count: u32,
+    console_path: String,
+}
+
+impl TryFrom<ConsoleArgs> for ConsoleConfiguration {
+    type Error = Error;
+
+    fn try_from(args: ConsoleArgs) -> Result<Self> {
+        if args.socket_count == 0 {
+            return Err(Error::SocketCountInvalid(0));
+        }
+
+        let console_path = args.console_path.trim().to_string();
+
+        Ok(ConsoleConfiguration {
+            socket_path: args.socket_path,
+            socket_count: args.socket_count,
+            console_path,
+        })
+    }
+}
+
+fn start_backend(args: ConsoleArgs) -> Result<()> {
+
+    let config = ConsoleConfiguration::try_from(args).unwrap();
+    let mut handles = Vec::new();
+
+    for _ in 0..config.socket_count {
+        let socket = config.socket_path.to_owned();
+        let console_path = config.console_path.to_owned();
+
+        let handle: JoinHandle<Result<()>> = spawn(move || loop {
+            // A separate thread is spawned for each socket and console connect to a separate guest.
+            // These are run in an infinite loop to not require the daemon to be restarted once a
+            // guest exits.
+            //
+            // There isn't much value in complicating code here to return an error from the
+            // threads, and so the code uses unwrap() instead. The panic on a thread won't cause
+            // trouble to other threads/guests or the main() function and should be safe for the
+            // daemon.
+
+            let controller = ConsoleController::new(console_path.clone())
+                .map_err(Error::CouldNotCreateConsoleController)?;
+            let arc_controller = Arc::new(RwLock::new(controller));
+            let vu_console_backend = Arc::new(RwLock::new(
+                VhostUserConsoleBackend::new(arc_controller)
+                    .map_err(Error::CouldNotCreateBackend)?,
+            ));
+
+            let mut daemon = VhostUserDaemon::new(
+                String::from("vhost-device-console-backend"),
+                vu_console_backend.clone(),
+                GuestMemoryAtomic::new(GuestMemoryMmap::new()),
+            )
+            .map_err(Error::CouldNotCreateDaemon)?;
+
+            /* Start the read thread -- need to handle it after termination */
+            let vring_workers = daemon.get_epoll_handlers();
+            vu_console_backend
+                .read()
+                .unwrap()
+                .set_vring_worker(&vring_workers[0]);
+            let read_handle = VhostUserConsoleBackend::start_console_thread(&vu_console_backend);
+
+            let listener = Listener::new(socket.clone(), true).unwrap();
```

Comment:
I think you meant to use `SOCKET_PATH{0,1,..., socket_count}` here instead of the generic `socket` path, right?